### ISSUE TITLE
The name field in twisted.conch.ssh.channel.SSHChannel should be a bytestring

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8683]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)| @einarfd reported|
|-|-|
|Trac ID|trac#8683|
|Type|enhancement|
|Created|2016-07-26 10:07:51Z|

The name field in SSHChannel should be a bytestring and the openChannel function in SSHConnection actually depends on it. 

<details><summary>Searchable metadata</summary>

```
trac-id__8683 8683
type__enhancement enhancement
reporter__einarfd einarfd
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469527671963816 1469527671963816
changetime__1469671985248094 1469671985248094
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)<a name="note_1"></a>|@einarfd removed owner|
|-|-|

Added merge request for this bug at https://github.com/twisted/twisted/pull/416

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset baba4c661a6606a56d8dab053cfe7b336a3c593e
```
#!CommitTicketReference repository="" revision="baba4c661a6606a56d8dab053cfe7b336a3c593e"
Merge pull request #435 from twisted/8683-ssh-channel-name-bytestring-1

Authord: einarfd
Reviewer: rodrigc
Fixes: #8683 

Convert ssh channel name to bytestring
```
